### PR TITLE
Quality checker upgrade and fix reported issues

### PR DESCRIPTION
**Pull request by zeshanziya**
Drupal Quality checker upgrade and related changes

### CODE REVIEW

File: grumphp.yml [hunk 1]
```diff
@@ -6,6 +6,19 @@ grumphp:
...
+        - web/modules/contrib
+        - web/themes/contrib
+        - web/profiles/contrib
+        - web/sites/default
+        - vendor
+      triggered_by:
+        - php
+        - module
+        - theme
+        - inc
```
Comment (hussainweb) [inline grumphp.yml:21]: This is duplicated.

File: grumphp.yml [hunk 2]
```diff
@@ -37,6 +50,7 @@ grumphp:
         - .github
         - .gitlab
         - .ddev
+        - .ddev
```
Comment (hussainweb) [inline grumphp.yml:53]: This is duplicated.

File: phpmd.xml [hunk 3]
```diff
@@ -9,26 +9,17 @@
   </description>
 
   <!-- Import common rulesets -->
-  <rule ref="rulesets/codesize.xml"/>
-  <rule ref="rulesets/design.xml"/>
+  <rule ref="rulesets/design.xml">
+    <exclude name="CouplingBetweenObjects"/>
```
Comment (hussainweb) [inline phpmd.xml:13]: I don't see this in DrupalQualityChecker.

File: .gitignore [hunk 4]
```diff
@@ -28,3 +28,6 @@
 node_modules
 /web/themes/custom/contribtracker/dist/
 .DS_Store
+
+# Ignore drupal-quality-checker generated config
+/*.dist
```
Comment (hussainweb) [inline .gitignore:33]: This is already in the file now at the top.

**Review by hussainweb — CHANGES_REQUESTED** [review]
I am not sure if this is a WIP PR. Still, sharing my comments.

**Review by hussainweb — CHANGES_REQUESTED** [review]
I am not sure if this is a WIP PR. Still, sharing my comments.

File: phpmd.xml (hunk 3 above)
Comment (zeshanziya) [reply phpmd.xml:13]: Yes It's not in the current release. I have raised a PR on quality checker for the same.

File: grumphp.yml (hunk 1 above)
Comment (zeshanziya) [reply grumphp.yml]: I think I messed it up while resolving merge conflict. I have removed the duplicate lines now.

File: .gitignore (hunk 4 above)
Comment (zeshanziya) [reply .gitignore]: Ahh missed it. Removed it

**Comment by zeshanziya** [comment]
> Last thing, can you revert changes to `composer.lock` here? There is an unintentional update but another problem is that it is changing the branch name for some path dependencies.

@hussainweb I have reverted `composer.lock` changes. Please review

**Review by hussainweb — APPROVED** [review]

**Pull request merged**